tests/unit/src/test/scala/tests/RenameFilesLspSuite.scala

```diff
@@ -588,55 +669,214 @@ class RenameFilesLspSuite extends BaseRenameFilesLspSuite("rename_files") {
   )
 
   renamed(
-    "static-obj".ignore,
-    s"""|/$prefix/A/Dep.scala
+    "qualified-refs-dir-move",
+    s"""|/$prefix/A/B/Sun.scala
+        |package <<A>>.B
+        |object Sun 
+        |/$prefix/X/Moon.scala
+        |package X
+        |
+        |object Moon {
+        |  <<A>>.B.Sun
+        |}
+        |""".stripMargin,
+    fileRenames = Map(s"$prefix/A" -> s"$prefix/C"),
+    expectedRenames = Map("A" -> "C"),
+    sourcesAreCompiled = true,
+  )
+
+  renamed(
+    "partially-qualified-names",
+    s"""|/$prefix/A/B/Sun.scala
+        |package A.<<B>>
+        |object Sun 
+        |/$prefix/X/Moon.scala
+        |package X
+        |import A.B
+        |<<//>>
+        |object Moon {
+        |  println(<<B.Sun>>)
+        |}
+        |""".stripMargin,
+    fileRenames = Map(s"$prefix/A/B" -> s"$prefix/A/C"),
+    expectedRenames =
+      Map("B" -> "C", "B.Sun" -> "Sun", "//" -> "import A.C.Sun\n//"),
+    sourcesAreCompiled = true,
+  )
+
+  renamed(
+    "top-level-declarations",
+    s"""|/$prefix/A/B/Sun.scala
         |package <<A>>
-        |object Obj {
-        | case class Dep()
+        |package object <<B>> {
+        | val m : Int = 3
         |}
-        |/$prefix/B/Usage.scala
+        |/$prefix/X/Moon.scala
+        |package X
+        |import <<A>>.<<B>>.m
+        |object Moon {
+        | val o = m
+        |}
+        |""".stripMargin,
+    fileRenames = Map(s"$prefix/A/B/Sun.scala" -> s"$prefix/C/D/Sun.scala"),
+    expectedRenames = Map("A" -> "C", "B" -> "D"),
+    sourcesAreCompiled = true,
+  )
+
+  renamed(
+    "implicits-scala3",
+    s"""|/$prefix/A/B/Sun.scala
+        |package <<A.B>>
+        |
+        |case class Sun()
+        |case class Moon()
+        |given m : Moon = Moon()
+        |
+        |extension(sun: Sun)
+        | def get2(using m : Moon) = 2
+        |/$prefix/X/Moon.scala
+        |package X
+        |import <<A.B.*
+        |import A.B.given>>
```

Comment:
So both when trying to delete wildcard import or given we check if this only imports things from that moved file/files, otherwise the import is not deleted. I'll make such a test for givens too. The heuristic for this does not distinguish between givens/other things. Would you like to `import C.D.given` instead of explicitly importing `import C.D.m` ? I change wildcards, givens to explicit imports, since otherwise it may import too much but I/m not sure if that is a good approach.

Reply:
I think it's better do `import C.D.given`